modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2925,8 +2957,18 @@ void GDScriptAnalyzer::reduce_identifier_from_base(GDScriptParser::IdentifierNod
 						break;
 				}
 			}
-			outer = outer->outer;
+			GDScriptParser::ClassNode *outer = script_class->outer;
+			while (outer != nullptr) {
+				script_classes.append(outer);
+				outer = outer->outer;
+			}
```

Comment:
i think maybe this and the next loop should be instead something like this:
```cpp
GDScriptParser::ClassNode *outer_base_class = script_class->base_type.class_type;
if (outer_base_class != nullptr) {
    script_classes.append(outer_base_class);
    GDScriptParser::ClassNode *outer_base_outer_class = outer_base_class->outer;
    while (outer_base_outer_class != nullptr) {
        script_classes.append(outer_base_outer_class );
        outer = outer_base_outer_class ->outer;
    }
}
```
and each run is one of the blue circles:
![image](https://user-images.githubusercontent.com/55224592/205528556-bdde1597-eab6-44b4-831d-1882aa6754f0.png)

Reply:
wait no, the bases of each outer need to be searched first...
i think i'm back to thinking recursion is the least complicated way to do this, and just have `checked_script_classes` as a member of the analyzer

Reply:
@rune-scape If you want to make a PR with your recursive changes and add me as a co-author, go for it! You seem to have an idea to fix this.